src/txn/txn_ckpt.c

```diff
@@ -249,23 +243,18 @@ __wt_checkpoint_list(WT_SESSION_IMPL *session, const char *cfg[])
 	if (F_ISSET(S2BT(session), WT_BTREE_NO_CHECKPOINT))
 		return (0);
 
-	/* Make sure there is space for the next entry. */
-	WT_RET(__wt_realloc_def(session, &session->ckpt_handle_allocated,
-	    session->ckpt_handle_next + 1, &session->ckpt_handle));
-
 	/* Not strictly necessary, but cleaner to clear the current handle. */
 	name = session->dhandle->name;
 	session->dhandle = NULL;
 
-	/* Record busy file names, we'll deal with them in the checkpoint. */
-	if ((ret = __wt_session_get_btree(session, name, NULL, NULL, 0)) == 0)
-		session->ckpt_handle[session->ckpt_handle_next++].dhandle =
-		    session->dhandle;
-	else if (ret == EBUSY)
-		ret = __wt_strdup(session, name,
-		    &session->ckpt_handle[session->ckpt_handle_next++].name);
+	if ((ret = __wt_session_get_btree(session, name, NULL, NULL, 0)) != 0)
+		return (ret == EBUSY ? 0 : ret);
 
-	return (ret);
+	/* Make sure there is space for the next entry. */
+	WT_RET(__wt_realloc_def(session, &session->ckpt_handle_allocated,
+	    session->ckpt_handle_next + 1, &session->ckpt_handle));
+	session->ckpt_handle[session->ckpt_handle_next++] = session->dhandle;
+	return (0);
```

Comment:
@michaelcahill 
To be clear, this isn't a change, but shouldn't we do the allocation before calling `__wt_session_get_btree`, that is, is it safe to leave the result of `__wt_session_get_btree` in the session handle?